plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -142,13 +144,40 @@ class CodeWhispererCodeScanManager(val project: Project) {
             return
         }
 
-        // Prepare for a code scan
-        beforeCodeScan()
+        //  If scanType is project
+        if (scanType == CodeWhispererConstants.SecurityScanType.PROJECT) {
+            // launch code scan coroutine
+            codeScanJob = launchCodeScanCoroutine(CodeWhispererConstants.SecurityScanType.PROJECT)
+        } else {
+            if (CodeWhispererExplorerActionManager.getInstance().isAutoEnabledForCodeScan()) {
+                // Prepare for a code scan
+                beforeCodeScan()
+                //  Add File Scan
+                codeScanJob = launchCodeScanCoroutine(CodeWhispererConstants.SecurityScanType.FILE)
+            }
+        }
```

Comment:
I think you have these the wrong way, right? Project scans should run `beforeCodeScan()`.

Reply:
Thanks for the catch